server: replace admin checks with VIEWCLUSTERMETADATA and REPAIRCLUSTERMETADATA

rafiss opened the pull request:
Release note (security update): Endpoints in the admin and status server
that previously required the admin role now can be used by users with
the VIEWCLUSTERMETADATA or REPAIRCLUSTERMETADATA system privilege,
depending on whether the endpoint is read-only or can modify state.

fixes https://github.com/cockroachdb/cockroach/issues/79571
fixes https://github.com/cockroachdb/cockroach/issues/109814
Release note: None

rafiss committed:
server: replace admin checks with VIEWCLUSTERMETADATA and REPAIRCLUSTERMETADATA

Release note (security update): Endpoints in the admin and status server
that previously required the admin role now can be used by users with
the VIEWCLUSTERMETADATA or REPAIRCLUSTERMETADATA system privilege,
depending on whether the endpoint is read-only or can modify state.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110609)
<!-- Reviewable:end -->

rafiss committed:
sql: remove usages of UserHasAdminRole

This commit covers a few cases that were missed by an earlier commit:
https://github.com/cockroachdb/cockroach/pull/110084. No release note is
included since it would be redundant with the release note from that PR.

Release note: None

dhartunian reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with some minor comments.

> Reviewed 8 of 8 files at r1, 1 of 8 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110609#-:-NexikjzB-XFTQqsroxG:brciwku)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ecwall, @rafiss, and @Santamaura)

___
*[`-- commits` line 10 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/110609#-NexcBUm18Lf4DMURNe4):*
does this need to have a release note noting the security change that allows the new fine-grained permissions?
___
*[`pkg/server/status.go` line 3637 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/110609#-NexbFsz-qVJbJMB7We5:-NexbFsz-qVJbJMB7We6:ban35cv) ([raw file](https://github.com/cockroachdb/cockroach/blob/b09d355a069f4a45298fdeabf232de8ddd723858/pkg/server/status.go#L3637)):*
> ```Go
> 	// if !isAdmin {
> 	// 	return nil, privchecker.ErrRequiresAdmin
> 	// }
> ```

nit: remove this?


<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110609)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian, @ecwall, and @Santamaura)

___
*[`-- commits` line 10 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/110609#-NexcBUm18Lf4DMURNe4):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

does this need to have a release note noting the security change that allows the new fine-grained permissions?
</blockquote></details>

i think the release note would be redundant with the one above and the one i added in https://github.com/cockroachdb/cockroach/pull/110084 so i opted to put no note here. i can put that in the commit message though.
___
*[`pkg/server/status.go` line 3637 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/110609#-NexbFsz-qVJbJMB7We5:-NexxCg9-lbPWkHinqPS:b-jcw1b6) ([raw file](https://github.com/cockroachdb/cockroach/blob/b09d355a069f4a45298fdeabf232de8ddd723858/pkg/server/status.go#L3637)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

nit: remove this?
</blockquote></details>

thanks for the catch!


<!-- Sent from Reviewable.io -->

rafiss commented:
TFTR!

bors r+
